packages/core/src/utils/delay.js

```diff
@@ -24,7 +24,7 @@ module.exports = function delay(func, wait) {
 
     if (waitingPromise === null) {
       waitingPromise = Promise.all([Promise.delay(wait), runningPromise])
```

Comment:
One question: Will this promise ever be rejected? If it does will this cause a change in behaviour (`.then` will not run in this case but `.finally` will always run regardless)?

Reply:
I don't think there is a case for that promise to be rejected. My observation is that `waitingPromise` is initially a `Promise.all` of a `Promise.delay` and a `Promise.resolve`, and based on the callback, subsequent `waitingPromise` will be either:

- Similar to the original (`Promise.all` of `Promise.delay` and `Promise.resolve`; or
- A `Promise.all` of `Promise.delay` and the *previous* `waitingPromise`

As the behaviour of `Promise.all` depends on the behaviour of the Promises inside, `waitingPromise` essentially just boils down to the behaviour of `Promise.delay` and `Promise.resolve`. Both have no chance of rejection: `Promise.delay` without second argument returns undefined on resolve, the same with `Promise.resolve`. So in the end, `waitingPromise` will never be rejected.

Reply:
Alright :) Thanks for the clarification and detailed explanation 👍